server/src/main/java/org/elasticsearch/action/admin/cluster/allocation/TransportGetAllocationStatsAction.java

```diff
@@ -63,21 +65,22 @@ public TransportGetAllocationStatsAction(
             threadPool.executor(ThreadPool.Names.MANAGEMENT)
         );
         this.allocationStatsService = allocationStatsService;
+        this.diskThresholdSettings = new DiskThresholdSettings(clusterService.getSettings(), clusterService.getClusterSettings());
     }
 
     @Override
     protected void doExecute(Task task, Request request, ActionListener<Response> listener) {
         if (clusterService.state().getMinTransportVersion().before(TransportVersions.ALLOCATION_STATS)) {
             // The action is not available before ALLOCATION_STATS
-            listener.onResponse(new Response(Map.of()));
+            listener.onResponse(new Response(Map.of(), diskThresholdSettings));
             return;
         }
         super.doExecute(task, request, listener);
     }
 
     @Override
     protected void masterOperation(Task task, Request request, ClusterState state, ActionListener<Response> listener) throws Exception {
-        listener.onResponse(new Response(allocationStatsService.stats()));
+        listener.onResponse(new Response(allocationStatsService.stats(), diskThresholdSettings));
```

Comment:
I would prefer to check `clusterService.state().getMinTransportVersion()` against `TransportVersions.WATERMARK_THRESHOLDS_STATS` here and pass a `null` if the cluster is not fully upgraded. If we don't, then we potentially drop this value on the transport layer (which tends to be trappy), but even if the values get all the way to the user on this request they may not on the next request and that is confusing and not in line with [these docs](https://www.elastic.co/guide/en/elastic-stack/8.13/upgrading-elasticsearch.html#rolling-upgrades):

> During a rolling upgrade, the cluster continues to operate normally. However, any new functionality is disabled or operates in a backward compatible mode until all nodes in the cluster are upgraded. New functionality becomes operational once the upgrade is complete and all nodes are running the new version.